pkg/setting/util.go

```diff
@@ -0,0 +1,100 @@
+package setting
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hobbyfarm/gargantua/v3/pkg/property"
+	settingProto "github.com/hobbyfarm/gargantua/v3/protos/setting"
+)
+
+const (
+	SettingRegistrationDisabled string = "registration-disabled"
+	SettingAdminUIMOTD          string = "motd-admin-ui"
+	SettingUIMOTD               string = "motd-ui"
+	ScheduledEventRetentionTime string = "scheduledevent-retention-time"
+)
+
+var DataTypeMappingToProto = map[property.DataType]settingProto.DataType{
+	property.DataTypeFloat:   settingProto.DataType_DATA_TYPE_FLOAT,
+	property.DataTypeInteger: settingProto.DataType_DATA_TYPE_INTEGER,
+	property.DataTypeString:  settingProto.DataType_DATA_TYPE_STRING,
+	property.DataTypeBoolean: settingProto.DataType_DATA_TYPE_BOOLEAN,
+}
+
+var ValueTypeMappingToProto = map[property.ValueType]settingProto.ValueType{
+	property.ValueTypeArray:  settingProto.ValueType_VALUE_TYPE_ARRAY,
+	property.ValueTypeMap:    settingProto.ValueType_VALUE_TYPE_MAP,
+	property.ValueTypeScalar: settingProto.ValueType_VALUE_TYPE_SCALAR,
+}
+
+var DataTypeMappingToHfTypes = map[settingProto.DataType]property.DataType{
+	settingProto.DataType_DATA_TYPE_FLOAT:   property.DataTypeFloat,
+	settingProto.DataType_DATA_TYPE_INTEGER: property.DataTypeInteger,
+	settingProto.DataType_DATA_TYPE_STRING:  property.DataTypeString,
+	settingProto.DataType_DATA_TYPE_BOOLEAN: property.DataTypeBoolean,
+}
+
+var ValueTypeMappingToHfTypes = map[settingProto.ValueType]property.ValueType{
+	settingProto.ValueType_VALUE_TYPE_ARRAY:  property.ValueTypeArray,
+	settingProto.ValueType_VALUE_TYPE_MAP:    property.ValueTypeMap,
+	settingProto.ValueType_VALUE_TYPE_SCALAR: property.ValueTypeScalar,
+}
+
+func FromJSON(p *settingProto.Property, value string) (any, error) {
```

Comment:
I am unsure of the role of this function.

Reply:
It is used in gargantua/services/settingsvc/internal/settingservice.go. And I implemented it because, we're now fetching the settings object from a grpc service instead of directly via the hfclient. The setting.Value, of our crd is of type string and we want to convert it to its represented golang type before marshaling it to JSON. Our frontend eventually expects a string, number, boolean, array or map.